A build error will occur because Copy and Clone are not implemented.

buildable
```rust
use kelp::{h2z, kata2hira};
use kelp::ConvOption;

fn convert_to_hiragana(text: &str) -> String {
    let options = ConvOption {
        ascii: true,
        digit: true,
        kana: true,
        ignore: "Aｱｶ0ｶﾞ".to_string(),
    };
    let mut hiragana = h2z(text, options);

    let options = ConvOption {
        ascii: true,
        digit: true,
        kana: true,
        ignore: "Aｱｶ0ｶﾞ".to_string(),
    };
    hiragana = kata2hira(&hiragana, options);
    hiragana
}

fn main() {
    println!("{}", convert_to_hiragana("ABCｱｲｳ123ｶﾞｶアー―−─━−ー〜~➖️ｰ〰️ガカ゛ハ゜パ"));
}
```
A build error will occur because Copy and Clone are not implemented.
```rust
use kelp::{h2z, kata2hira};
use kelp::ConvOption;

fn convert_to_hiragana(text: &str) -> String {
    let options = ConvOption {
        ascii: true,
        digit: true,
        kana: true,
        ignore: "Aｱｶ0ｶﾞ".to_string(),
    };
    let mut hiragana = h2z(text, options);

    hiragana = kata2hira(&hiragana, options);
    hiragana
}

fn main() {
    println!("{}", convert_to_hiragana("ABCｱｲｳ123ｶﾞｶアー―−─━−ー〜~➖️ｰ〰️ガカ゛ハ゜パ"));
}
```
```
error[E0382]: use of moved value: `options`
  --> src/main.rs:13:37
   |
5  |     let options = ConvOption {
   |         ------- move occurs because `options` has type `ConvOption`, which does not implement the `Copy` trait
...
11 |     let mut hiragana = h2z(text, options);
   |                                  ------- value moved here
12 |
13 |     hiragana = kata2hira(&hiragana, options);
   |                                     ^^^^^^^ value used here after move

For more information about this error, try `rustc --explain E0382`.
error: could not compile `langdetect` (bin "langdetect") due to 1 previous error
```
Proposed fix
The handling of ignore is slightly different.
```diff
diff --git a/src/lib.rs b/src/lib.rs
index 6a46d37..b222d42 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -11,10 +11,10 @@ pub use convert::kata2hira;
 pub use convert::z2h;
 
 /// Convert options
-#[derive(Debug, Default)]
+#[derive(Debug, Default, Clone, Copy)]
 pub struct ConvOption {
     pub ascii: bool,
     pub digit: bool,
-    pub ignore: String,
+    pub ignore: &'static str,
     pub kana: bool,
 }
```
Buildable
```rust
use kelp::{h2z, kata2hira};
use kelp::ConvOption;

fn convert_to_hiragana(text: &str) -> String {
    let options = ConvOption {
        ascii: true,
        digit: true,
        kana: true,
        ignore: "Aｱｶ0ｶﾞ",
    };
    let mut hiragana = h2z(text, options);

    hiragana = kata2hira(&hiragana, options);
    hiragana
}

fn main() {
    println!("{}", convert_to_hiragana("ABCｱｲｳ123ｶﾞｶアー―−─━−ー〜~➖️ｰ〰️ガカ゛ハ゜パ"));
}
```

--- a/src/bin/kelp.rs
+++ b/src/bin/kelp.rs
@@ -88,10 +88,14 @@ struct Args {
 
 fn main() {
     let args = Args::parse();
+    let ignore: &'static str = match args.ignore.as_deref() {
+        Some("") | None => "",
+        Some(s) => Box::leak(s.to_string().into_boxed_str()),
+    };
     let option = ConvOption {
         ascii: args.ascii,
         digit: args.digit,
-        ignore: args.ignore.unwrap_or("".to_string()),
+        ignore: ignore,
         kana: args.kana,
     };
     let text = args.text.as_deref().unwrap_or("");

--- a/src/bin/kelp.rs
+++ b/src/bin/kelp.rs
@@ -88,7 +88,7 @@ struct Args {
 
 fn main() {
     let args = Args::parse();
-    let ignore: &'static str = match args.ignore.as_deref() {
+    let ignore: &str = match args.ignore.as_deref() {
         Some("") | None => "",
         Some(s) => Box::leak(s.to_string().into_boxed_str()),
     };

--- a/src/conv_table.rs
+++ b/src/conv_table.rs
@@ -209,7 +209,7 @@ pub(crate) enum Target {
     Kana,
 }
 
-impl From<&ConvOption> for Target {
+impl From<&ConvOption<'_>> for Target {
     fn from(option: &ConvOption) -> Target {
         match (option.ascii, option.digit, option.kana) {
             (true, true, true) => All,

--- a/src/convert.rs
+++ b/src/convert.rs
@@ -20,7 +20,7 @@ use crate::ConvOption;
 /// assert_eq!("アイウエオ", converted);
 ///
 /// let option = ConvOption {
-///     ignore: "かこ".to_string(),
+///     ignore: "かこ",
 ///     ..Default::default()
 /// };
 /// let converted = hira2kata("かきくけこ", option);
@@ -46,7 +46,7 @@ pub fn hira2kata(text: &str, option: ConvOption) -> String {
 /// assert_eq!("ｱｲｳｴｵ", converted);
 ///
 /// let option = ConvOption {
-///     ignore: "がご".to_string(),
+///     ignore: "がご",
 ///     ..Default::default()
 /// };
 /// let converted = hira2hkata("がぎぐげご", option);
@@ -72,7 +72,7 @@ pub fn hira2hkata(text: &str, option: ConvOption) -> String {
 /// assert_eq!("あいうえお", converted);
 ///
 /// let option = ConvOption {
-///     ignore: "キクケ".to_string(),
+///     ignore: "キクケ",
 ///     ..Default::default()
 /// };
 /// let converted = kata2hira("カキクケコ", option);
@@ -104,7 +104,7 @@ pub fn kata2hira(text: &str, option: ConvOption) -> String {
 ///     ascii: true,
 ///     digit: true,
 ///     kana: true,
-///     ignore: "Aｱ0".to_string(),
+///     ignore: "Aｱ0",
 /// };
 /// let converted = h2z("ABCｱｲｳ012", option);
 /// assert_eq!("AＢＣｱイウ0１２", converted);
@@ -142,7 +142,7 @@ pub fn h2z(text: &str, option: ConvOption) -> String {
 /// let option = ConvOption {
 ///     ascii: true,
 ///     digit: true,
-///     ignore: "Ａア０".to_string(),
+///     ignore: "Ａア０",
 ///     kana: true,
 /// };
 /// let converted = z2h("ＡＢＣアイウ０１２", option);
@@ -207,7 +207,7 @@ mod tests {
     fn test_hira2kata_with_ignore() {
         let before = strings!(HIRAGANA);
         let option = ConvOption {
-            ignore: before.clone(),
+            ignore: &before,
             ..Default::default()
         };
         assert_eq!(hira2kata(&before, option), before);
@@ -227,7 +227,7 @@ mod tests {
     fn test_hira2hkata_with_ignore() {
         let before = strings!(HIRAGANA);
         let option = ConvOption {
-            ignore: before.clone(),
+            ignore: &before,
             ..Default::default()
         };
         assert_eq!(hira2hkata(&before, option), before);
@@ -247,7 +247,7 @@ mod tests {
     fn test_kata2hira_with_ignore() {
         let before = strings!(FULL_KANA);
         let option = ConvOption {
-            ignore: before.clone(),
+            ignore: &before,
             ..Default::default()
         };
         assert_eq!(kata2hira(&before, option), before);

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -12,9 +12,9 @@ pub use convert::z2h;
 
 /// Convert options
 #[derive(Debug, Default, Clone, Copy)]
-pub struct ConvOption {
+pub struct ConvOption<'a> {
     pub ascii: bool,
     pub digit: bool,
-    pub ignore: &'static str,
+    pub ignore: &'a str,
     pub kana: bool,
 }

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -11,10 +11,10 @@ pub use convert::kata2hira;
 pub use convert::z2h;
 
 /// Convert options
-#[derive(Debug, Default)]
+#[derive(Debug, Default, Clone, Copy)]
 pub struct ConvOption {
     pub ascii: bool,
     pub digit: bool,
-    pub ignore: String,
+    pub ignore: &'static str,
     pub kana: bool,
 }

